### PR TITLE
Perform batch decryption in wallet scanning (legacy poc branch)

### PR DESCRIPTION
**Pull request by nuttycom**
At present, we batch on a per-block basis.

### CODE REVIEW

File: components/zcash_note_encryption/src/batch.rs [hunk 1]
```diff
@@ -14,7 +12,7 @@ use crate::{
 pub fn try_note_decryption<D: BatchDomain, Output: ShieldedOutput<D>>(
     ivks: &[D::IncomingViewingKey],
     outputs: &[(D, Output)],
-) -> Vec<Option<(D::Note, D::Recipient, D::Memo)>> {
+) -> Vec<Option<((D::Note, D::Recipient, D::Memo), usize)>> {
```
Comment (daira) [inline components/zcash_note_encryption/src/batch.rs:15]: Explain this in the doc comment.

File: components/zcash_note_encryption/src/batch.rs [hunk 2]
```diff
@@ -24,25 +22,29 @@ pub fn try_note_decryption<D: BatchDomain, Output: ShieldedOutput<D>>(
 pub fn try_compact_note_decryption<D: BatchDomain, Output: ShieldedOutput<D>>(
     ivks: &[D::IncomingViewingKey],
     outputs: &[(D, Output)],
-) -> Vec<Option<(D::Note, D::Recipient)>> {
+) -> Vec<Option<((D::Note, D::Recipient), usize)>> {
```
Comment (daira) [inline components/zcash_note_encryption/src/batch.rs:25]: Same here.

**Comment by pacu** [comment]
I tested this on an internal Build of ECC wallet. and seems ok. Improvement is not noticeable to the naked eye but any enhancement is very welcome 🙂

**Comment by nuttycom** [comment]
Closing as it seems likely we won't make another wallet release based on the legacy POC branch.

**Pull request closed without merging**